packages/medusa/src/strategies/batch-jobs/product/export.ts

```diff
@@ -0,0 +1,393 @@
+
+  private appendOptionsDescriptors(maxOptionsCount: number): void {
+    for (let i = 0; i < maxOptionsCount; ++i) {
+      this.columnDescriptors
+        .set(`Option ${i + 1} Name`, {
+          accessor: (productOption: Product) =>
+            productOption?.options[i]?.title ?? "",
+          entityName: "product",
+        })
+        .set(`Option ${i + 1} Value`, {
+          accessor: (variant: ProductVariant) =>
+            variant?.options[i]?.value ?? "",
+          entityName: "variant",
+        })
+    }
+  }
+
+  private appendMoneyAmountDescriptors(maxMoneyAmountCount: number): void {
+    for (let i = 0; i < maxMoneyAmountCount; ++i) {
+      this.columnDescriptors
+        .set(`Price ${i + 1} Currency code`, {
+          accessor: (variant: ProductVariant) =>
+            variant?.prices[i]?.currency_code?.toLowerCase() ?? "",
+          entityName: "variant",
+        })
+        .set(`Price ${i + 1} Region name`, {
+          accessor: (variant: ProductVariant) =>
+            variant?.prices[i]?.region?.name?.toLowerCase() ?? "",
+          entityName: "variant",
+        })
+        .set(`Price ${i + 1} Amount`, {
+          accessor: (variant: ProductVariant) =>
+            variant?.prices[i]?.amount?.toString() ?? "",
+          entityName: "variant",
+        })
+    }
+  }
+
+  private buildProductVariantLines(product: Product): string[] {
+    const outputLineData: string[] = []
+
+    for (const variant of product.variants) {
+      const variantLineData: string[] = []
+      for (const [, columnSchema] of this.columnDescriptors.entries()) {
+        if (columnSchema.entityName === "product") {
+          variantLineData.push(columnSchema.accessor(product))
+        }
+        if (columnSchema.entityName === "variant") {
+          variantLineData.push(columnSchema.accessor(variant))
+        }
+      }
+      variantLineData.push(this.NEWLINE_)
+      outputLineData.push(variantLineData.join(this.DELIMITER_))
+    }
+
+    return outputLineData
+  }
+
+  private async handleProcessingErrors(
+    batchJobId: string,
+    err: unknown,
+    {
+      offset,
+      count,
+      advancementCount,
+      progress,
+    }: {
+      offset: number
+      count: number
+      advancementCount: number
+      progress: number
+    }
+  ): Promise<void> {
+    return await this.atomicPhase_(async (transactionManager) => {
+      const batchJob = await this.batchJobService_
+        .withTransaction(transactionManager)
+        .retrieve(batchJobId)
+
+      const retryCount = batchJob.context.retry_count ?? 0
+      const maxRetry = batchJob.context.max_retry ?? this.defaultMaxRetry
+
+      const errMessage =
+        (err as any).message ??
+        `Something went wrong with the batchJob ${batchJob.id}`
+
+      if (err instanceof MedusaError) {
+        await this.batchJobService_
+          .withTransaction(transactionManager)
+          .setFailed(batchJob, errMessage)
+      } else if (retryCount < maxRetry) {
```

Comment:
Yeah, that part is a bit experimental and still required some thinking I think

Reply:
I think the retry could be handled by the abstract class to make it consistent across the strategies, and it would just require to implement a method on the child class that would be something like shouldRetry on error and get the error as well as the job in the arguments, wdyt?